tests/search_tests.rs

```diff
@@ -39,6 +40,14 @@ fn test_build_superstructure_query() {
     );
 }
 
+#[test]
+fn test_build_similarity_query() {
+    let ranked_clusters = vec![0, 2];
+    let extra_query = "NumAtoms:[1 TO 100] AND NumAmideBonds: [1 TO 5]";
+    let query = build_similarity_query(&ranked_clusters, extra_query, 0.1);
+    assert_eq!(query, "(other_descriptors.similarity_cluster:0 OR other_descriptors.similarity_cluster:2) AND NumAtoms:[1 TO 100] AND NumAmideBonds: [1 TO 5]");
+}
```

Comment:
Any chance we could get a little test of indexing documents X, Y, Z and then confirming the similarity search works a certain way?